chore(12): merge develop into release/12.0.0

emilyrohrbough opened the pull request:
merge develop into release/12.0.0

emilyrohrbough committed:
chore: forked pr is pull not pulls (#24840)

emilyrohrbough committed:
chore: trigger circleci when pr is ready workflow (#24841)

mjhenkes committed:
chore: disable commercial recommendations for system tests. Having them enabled causes failed test for contributor PRs (#24878)

emilyrohrbough committed:
Merge branch 'develop' into release/12.0.0

mike-plummer committed:
feat: Handle special characters in Create from Component filenames (#24864)

flotwig committed:
chore: remove stray debugger statement (#24899)

flotwig committed:
test: remove failing domain_spec system test (#24901)

marktnoonan committed:
chore: add component testing to testing types in CONTRIBUTING.md (#24908)

chrisbreiding committed:
feat: Add new local and session storage commands (#24859)

emilyrohrbough committed:
Merge branch 'release/12.0.0' of https://github.com/cypress-io/cypress into release/12.0.0

emilyrohrbough committed:
Merge branch 'develop' into 12/merge-dev

# Conflicts:
#	packages/driver/cypress/e2e/commands/local_storage.cy.js
#	packages/driver/cypress/e2e/e2e/origin/commands/misc.cy.ts
#	packages/driver/src/cy/commands/storage.ts

emilyrohrbough commented:
closing as dup of: https://github.com/cypress-io/cypress/pull/24926

ZachJW34 committed:
fix: add top-level run-all button to allow all specs to be run at once (#24846)

Co-authored-by: Lachlan Miller <lachlan.miller.1990@outlook.com>
Co-authored-by: Mark Noonan <mark@cypress.io>

BlueWinds commented:
Looks like some `experimentalSessionAndOrigin` leaked into the PR, failing check-ts.

emilyrohrbough committed:
remove reference to experimentalSessionAndOrigin

mjhenkes committed:
chore: disable commercial recommendations for system tests. Having them enabled causes failed test for contributor PRs. Take 2 (#24884)

abcfy2 committed:
fix: get correct env from npm config (#24664)

* fix: get correct env from npm config

`npm config set VAR VAL` will inject `npm_config_var=val` environment
variable. This commit will solve this issue

Closes: #24556

* Disable commercial recommendations for system tests.

* Disable commercial recommendations

* Update system-tests/lib/system-tests.ts

* Revert changes

* Update cli/test/lib/util_spec.js

* Update cli/test/lib/util_spec.js

Co-authored-by: Matt Henkes <mjhenkes@gmail.com>

mjhenkes committed:
Merge branch 'release/12.0.0' into 12/merge-dev

mjhenkes committed:
Merge branch 'develop' into 12/merge-dev

mjhenkes committed:
remove check for session and origin

emilyrohrbough committed:
Merge branch '12/merge-dev' of https://github.com/cypress-io/cypress into 12/merge-dev

emilyrohrbough committed:
update snapshot
